// @pulumi/kubernetes/helm/v3/release.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Release = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A Release is an instance of a chart running in a Kubernetes cluster.
 *
 * A Chart is a Helm package. It contains all of the resource definitions necessary to run an application, tool, or service inside of a Kubernetes cluster.
 */
class Release extends pulumi.CustomResource {
    /**
     * Create a Release resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.chart === undefined) && !opts.urn) {
                throw new Error("Missing required property 'chart'");
            }
            if ((!args || args.repositoryOpts === undefined) && !opts.urn) {
                throw new Error("Missing required property 'repositoryOpts'");
            }
            inputs["atomic"] = args ? args.atomic : undefined;
            inputs["chart"] = args ? args.chart : undefined;
            inputs["cleanupOnFail"] = args ? args.cleanupOnFail : undefined;
            inputs["compat"] = "true";
            inputs["createNamespace"] = args ? args.createNamespace : undefined;
            inputs["dependencyUpdate"] = args ? args.dependencyUpdate : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["devel"] = args ? args.devel : undefined;
            inputs["disableCRDHooks"] = args ? args.disableCRDHooks : undefined;
            inputs["disableOpenapiValidation"] = args ? args.disableOpenapiValidation : undefined;
            inputs["disableWebhooks"] = args ? args.disableWebhooks : undefined;
            inputs["forceUpdate"] = args ? args.forceUpdate : undefined;
            inputs["keyring"] = args ? args.keyring : undefined;
            inputs["lint"] = args ? args.lint : undefined;
            inputs["manifest"] = args ? args.manifest : undefined;
            inputs["maxHistory"] = args ? args.maxHistory : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["namespace"] = args ? args.namespace : undefined;
            inputs["postrender"] = args ? args.postrender : undefined;
            inputs["recreatePods"] = args ? args.recreatePods : undefined;
            inputs["renderSubchartNotes"] = args ? args.renderSubchartNotes : undefined;
            inputs["replace"] = args ? args.replace : undefined;
            inputs["repositoryOpts"] = args ? args.repositoryOpts : undefined;
            inputs["resetValues"] = args ? args.resetValues : undefined;
            inputs["resourceNames"] = args ? args.resourceNames : undefined;
            inputs["reuseValues"] = args ? args.reuseValues : undefined;
            inputs["skipAwait"] = args ? args.skipAwait : undefined;
            inputs["skipCrds"] = args ? args.skipCrds : undefined;
            inputs["timeout"] = args ? args.timeout : undefined;
            inputs["valueYamlFiles"] = args ? args.valueYamlFiles : undefined;
            inputs["values"] = args ? args.values : undefined;
            inputs["verify"] = args ? args.verify : undefined;
            inputs["version"] = args ? args.version : undefined;
            inputs["waitForJobs"] = args ? args.waitForJobs : undefined;
            inputs["status"] = undefined /*out*/;
        }
        else {
            inputs["atomic"] = undefined /*out*/;
            inputs["chart"] = undefined /*out*/;
            inputs["cleanupOnFail"] = undefined /*out*/;
            inputs["createNamespace"] = undefined /*out*/;
            inputs["dependencyUpdate"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["devel"] = undefined /*out*/;
            inputs["disableCRDHooks"] = undefined /*out*/;
            inputs["disableOpenapiValidation"] = undefined /*out*/;
            inputs["disableWebhooks"] = undefined /*out*/;
            inputs["forceUpdate"] = undefined /*out*/;
            inputs["keyring"] = undefined /*out*/;
            inputs["lint"] = undefined /*out*/;
            inputs["manifest"] = undefined /*out*/;
            inputs["maxHistory"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["namespace"] = undefined /*out*/;
            inputs["postrender"] = undefined /*out*/;
            inputs["recreatePods"] = undefined /*out*/;
            inputs["renderSubchartNotes"] = undefined /*out*/;
            inputs["replace"] = undefined /*out*/;
            inputs["repositoryOpts"] = undefined /*out*/;
            inputs["resetValues"] = undefined /*out*/;
            inputs["resourceNames"] = undefined /*out*/;
            inputs["reuseValues"] = undefined /*out*/;
            inputs["skipAwait"] = undefined /*out*/;
            inputs["skipCrds"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["timeout"] = undefined /*out*/;
            inputs["valueYamlFiles"] = undefined /*out*/;
            inputs["values"] = undefined /*out*/;
            inputs["verify"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
            inputs["waitForJobs"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Release.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Release resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Release(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Release.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Release.__pulumiType;
    }
}
exports.Release = Release;
/** @internal */
Release.__pulumiType = 'kubernetes:helm.sh/v3:Release';
